utils/build_rpms_from_scratch.sh

```diff
@@ -6,52 +6,126 @@ set -uex
+
+  if [[ "$OS_ID" == "opensuse-leap" ]]; then
+   dependencies=$(rpmspec -q --buildrequires "$spec_file" | grep -v "^rpmlib" | grep -v "^/" | tr '\n' ' ')
+      if [ -n "$dependencies" ]; then
+          # Install dependencies using zypper
+          echo "Installing build dependencies: $dependencies"
+          zypper install --allow-vendor-change -y $dependencies
+      else
+          echo "No build dependencies found in the spec file."
+      fi
+  else
+    dnf --refresh builddep -y "$spec_file"
+  fi
+}
+
+prepare_env_leap() {
+  suse_version=$OS_VERSION # default to '15.4'
+
+  # Update and install required packages using zypper
+  zypper refresh
+
+  # Install needed packages
+  # git so can clone the repo
+  # rpm-build, rpmdevtools, createrepo for building the rpms and setting up the rpm repo
+  # make is used for managing the rpmbuild process. This should probably be a dependency of the rpm.spec files, but not
+  # python3-Sphinx is needed by dpdk. The file /usr/bin/sphinx-build is listed as a dependency in the spec file,
+  #   however, zypper isn't able to resolve a filename to a package and the scripting to handle that case is pretty
+  #   messy, especially because "zypper what-provides" actually returns three options for sphinx-build. So, forgoing
+  #   automation here and simply hardcoding the dependency.
+  zypper install -y \
+      git \
+      rpm-build \
+      rpmdevtools \
+      createrepo_c \
+      make \
+      python3-Sphinx
+
+
+  # Install build tools and dependencies
+  zypper install -y -t pattern devel_basis
+
+  # Enable additional repositories if needed
+  if ! zypper lr | grep -q "Main Repository"; then
+      zypper ar -f http://download.opensuse.org/distribution/leap/$suse_version/repo/oss/ "Main Repository"
+  else
+      echo "Main Repository already exists, skipping addition."
+  fi
+  if ! zypper lr | grep -q "Main Update Repository"; then
+      zypper ar -f http://download.opensuse.org/update/leap/$suse_version/oss/ "Main Update Repository"
+  else
+      echo "Main Update Repository already exists, skipping addition."
+  fi
+
+  # Ignore dpdk because want to use RPMs built from daos-stack/dpdk
+  zypper update -y --skip-interactive --no-recommends
 }
 
-setup_rpm_build_env() {
-  el_version=${1:-8} # default to '8'
+prepare_env_rocky() {
+  el_version=$OS_VERSION # default to '8'
 
-  # Update and install required packages
+    # Install needed packages
+    # git so can clone the repo
+    # rpm-build, rpmdevtools, createrepo for building the rpms and setting up the rpm repo
+    # dnf-plugins-core for additional dnf capabilities
+    # make is used for managing the rpmbuild process. This should probably be a dependency of the rpm.spec files, but not
   dnf install -y \
       git \
       rpm-build \
       rpmdevtools \
-      dnf-plugins-core \
       createrepo \
+      dnf-plugins-core \
       epel-release \
       make # shouldn't be needed here, but dependency spec files don't include
```

Comment:
Actually it is required here since you are using `make rpms` which is independent of any requirement for `make` to build an RPM.